front/app/containers/Authentication/SignUpIn/index.tsx

```diff
@@ -21,18 +24,44 @@ const SignUpInContainer = ({ authUser, onModalOpenedStateChange }: Props) => {
 
   const { pathname, search } = useLocation();
 
-  useEffect(() => {
-    const isAuthError = endsWith(pathname, 'authentication-error');
-    const isInvitation = endsWith(pathname, '/invite');
+  const isAuthError = endsWith(pathname, 'authentication-error');
+  const isInvitation = endsWith(pathname, '/invite');
 
+  useEffect(() => {
     openSignUpInModalIfNecessary(
       authUser,
       isAuthError && !signUpInModalClosed,
       isInvitation && !signUpInModalClosed,
       signUpInModalMounted,
       search
     );
-  }, [pathname, search, authUser, signUpInModalClosed, signUpInModalMounted]);
+  }, [
+    isInvitation,
+    isAuthError,
+    search,
+    authUser,
+    signUpInModalClosed,
+    signUpInModalMounted,
+  ]);
+
+  // In case of a sign up / in route, open modal and redirect to homepage
+  useEffect(() => {
+    let timeout: NodeJS.Timeout;
+    const isSignInRoute = endsWith(pathname, 'sign-in');
+    const isSignUpRoute = endsWith(pathname, 'sign-up');
+    const isAuthRoute = isSignInRoute || isSignUpRoute;
+    if (isAuthRoute && isNilOrError(authUser)) {
+      timeout = setTimeout(() => {
```

Comment:
This is a little horrible but needed for the modal to open and then redirect to the homepage - open to better suggestions for it though.

Reply:
Let's add a comment?